src/PrestaShopBundle/Form/Admin/Configure/AdvancedParameters/Import/ImportFormDataProvider.php

```diff
@@ -45,23 +48,50 @@
      */
     private $importFileFinder;
 
-    public function __construct(SessionInterface $session, FileFinder $importFileFinder)
+    /**
+     * @var null|Request current request
+     */
+    private $request;
+
+    /**
+     * @param SessionInterface $session
+     * @param FileFinder $importFileFinder
+     * @param RequestStack $requestStack
+     */
+    public function __construct(SessionInterface $session, FileFinder $importFileFinder, RequestStack $requestStack)
     {
         $this->session = $session;
         $this->importFileFinder = $importFileFinder;
+        $this->request = $requestStack->getCurrentRequest();
```

Comment:
Like above: I dont feel god about injecting session and request into a SF service. I think it's because you tried to fit into  `FormDataProviderInterface` however we are now aware it does not fit all kind of forms.

We now have decided on another Interface for such Forms: see https://github.com/PrestaShop/PrestaShop/issues/10285#issuecomment-427765924

Reply:
I see now we already discussed that and we said it will be done in another PR.

Reply:
@matks I've thought about this issue with session/request injection in the form data provider, also discussed it with @sarjon and we couldn't see what's wrong with it. Since the data provider is always used when form is being built in a controller's context, the session and request will always be available there.
Reworking it to avoid injection of session/request would require a completely custom form handler (without following the `FormHandlerInterface`) and custom form data provider as well, but we don't see any pros of having it. Wdyt? :thinking:

Reply:
Here's a tough question 😄 I tried to find all the reasons I can think of. Here they are:

The 1st and main reason not to inject the request in a Symfony service is if this service is used in a CLI context. However, as you said, it is unlikely here as this is a Form data provider. So here this reason is not valid.

2nd reason is that it creates coupling between this class and HTTP requests. Although this is not totally unrelated because we're talking about forms here, it is better to try to isolate those matters as it usually improves the robustness and maintainability of a codebase. Imagine if tomorrow we create a PdfToHtmlWriter service and we inject the HTTP request as a constructor argument there. Does it look right ? HTML is mainly related to HTTP requests, but do you feel good, as a developer, if you encounter this ?

3rd reason is that we forbid the request to be passed "outside" the controller because we do not want that, in the future, a developer comes in and think "mmm ... OK I have access in this class to the request, so no need to do any processing in the controller, I don't care". This would be a bad developer but anyway this would lead to a situation where we break the "controller-layer" system: "controller handles the HTTP data, then pass righfully formatted data to underlying services". We would create the possibility to easily bypass the controller. If I play devil's advocate, I would say this is how we get `$_POST` and `$_GET` calls inside some deep/core services 😅 

4th reason is that by injecting the request into the Symfony service, you make this service dependent on the request structure. If this request carries a GET parameter `foo` and a POST parameter `bar`, we make the service aware of it as it now extracts there from the request. If tomorrow we change the HTTP request and now parameters are named `arg_foo` and `arg_bar` we have to update the Symfony service as well. Again this does not feel right: the structure of the data being exchanged between the browser and the Symfony controller is an interface between these two. Underlying services should not depend on it. If you convert a form input IDs from camelCase to snake_case, you do not expect your services to break, you expect your controllers and forms to break. 

5th reason and last reason is for testing: if we introduce as we want to (but could not find time so far to do it) functionnal CLI testings, we're going to boot Symfony kernel and start calling the services inside to send specific inputs and check we get the right outputs. Having the request as a dependency, we will not be able to perform this kind of test on this data provider.

Reply:
fair enough :smile: